src/widgets/modals/CashTransactionModal.js

```diff
@@ -19,38 +19,43 @@ import { PageButton } from '../PageButton';
 import { PencilIcon, ChevronDownIcon } from '../icons';
 
 import globalStyles, { WARM_GREY, SUSSOL_ORANGE } from '../../globalStyles';
+import { ToggleBar } from '../ToggleBar';
 
 const placeholderName = 'Choose a name';
-const placeholderType = 'Choose a transaction type';
 const placeholderAmount = 'Enter transaction amount';
 const placeholderReason = 'Choose a reason';
+const placeholderReasonDisabled = 'N/A';
 const placeholderDescription = 'Enter a description';
 
 export const CashTransactionModal = ({ onConfirm }) => {
   const [name, setName] = useState(null);
-  const [type, setType] = useState(null);
   const [amount, setAmount] = useState(null);
   const [reason, setReason] = useState(null);
   const [description, setDescription] = useState(null);
 
+  const [isCashIn, setIsCashIn] = useState(true);
   const [textBuffer, setTextBuffer] = useState('');
+
   const [isNameModalOpen, setIsNameModalOpen] = useState(false);
-  const [isTypeModalOpen, setIsTypeModalOpen] = useState(false);
   const [isAmountModalOpen, setIsAmountModalOpen] = useState(false);
   const [isReasonModalOpen, setIsReasonModalOpen] = useState(false);
   const [isDescriptionModalOpen, setIsDescriptionModalOpen] = useState(false);
 
   const names = useMemo(() => UIDatabase.objects('Name'));
   const reasons = useMemo(() => UIDatabase.objects('Options'));
+  const type = useMemo(
+    () => (isCashIn ? CASH_TRANSACTION_TYPES.CASH_IN : CASH_TRANSACTION_TYPES.CASH_OUT),
+    [isCashIn]
+  );
 
-  const isValidTransaction = useMemo(() => !!name && !!type && !!amount && !!reason, [
+  const isValidTransaction = useMemo(() => !!name && !!amount && (isCashIn || !!reason), [
     name,
-    type,
     amount,
     reason,
+    isCashIn,
```

Comment:
good selector. Also I'd imagine potentially just as fast to evaluate the expression each render rather than evaluate if each of the dependencies are equal, and then do the lookup in the cache potentially missing etc etc